### PR TITLE
Improve the Database Manager

### LINKED ISSUE #602: cannot update a table when one field is unique
- Laravel Version: 5.3.#
- Voyager Version: 0.10.11
- PHP Version: 7.1
- Database Driver & Version:

### Description:
When updating a table where one or more columns have 'unique' an SQL exception is thrown

### Steps To Reproduce:

create a new table: MyTest
id 
name - set key to Unique

save

edit the table, do not do anything, just hit 'Update table' and the following error occurs

SQLSTATE[42000]: Syntax error or access violation: 1061 Duplicate key name 'mytest_name_unique' (SQL: alter table `MyTest` add unique `mytest_name_unique`(`name`))

### PR DESCRIPTION
**Pull request by abdgad**
The goal of this is to improve database related tasks.
Still a work in progress...

### CODE REVIEW

**Comment by abdgad** [comment]
There is still a problem... working on it.

**Comment by marktopper** [comment]
TravisCI fails, do you have any ideas why?

**Review by Frondor — APPROVED** [review]
+1 for the corrections.

**Comment by Frondor** [comment]
That's weird... According to [Illuminate/Foundation/helpers.php#L154](https://github.com/laravel/framework/blob/5.3/src/Illuminate/Foundation/helpers.php#L154) it should be returing `\Illuminate\Http\RedirectResponse` instead of `\Illuminate\Http\Response` like TravisCI says... That guy is acting weird lately

**Comment by abdgad** [comment]
I am still working on this.
@Frondor it's throwing that exception because I changed update() to redirect back to the previous page, which is /database/table/edit instead of /database index

**Comment by abdgad** [comment]
@Frondor You are right, I am sorry for misunderstanding you about the Exception.
back() returns RedirectResponse indeed. Will dig deeper into this later.

File: resources/views/tools/database/edit-add.blade.php [hunk 1]
```diff
@@ -295,24 +295,24 @@ function newRow(kind, name, type, nullable, key, defaultValue) {
                 $('#' + unique_id).find('.fieldKey').hide();
                 $('#newFieldSoftDelete').attr('disabled', 'disabled').off('click');
             } else {
-                if (typeof(name) != 'undefined') {
+                if (typeof(row) != 'undefined') {
                     $('#' + unique_id).addClass('existing_row');
```
Comment (fergthh) [inline resources/views/tools/database/edit-add.blade.php:299]: You can caches the reference in order to improve script performance.

**Comment by marktopper** [comment]
Please ping me once this is no longer work in progress, then I will test it out and merge.

Also, really great work! 👍

**Comment by abdgad** [comment]
This has been closed and replaced by #623.

**Pull request closed without merging**